mcache.go

```diff
@@ -28,13 +28,19 @@ func NewMessageCache(gossip, history int) *MessageCache {
 		msgs:    make(map[string]*pb.Message),
 		history: make([][]CacheEntry, history),
 		gossip:  gossip,
+		msgID:   DefaultMsgIdFn,
 	}
 }
 
 type MessageCache struct {
 	msgs    map[string]*pb.Message
 	history [][]CacheEntry
 	gossip  int
+	msgID   MsgIdFunction
+}
+
+func (mc *MessageCache) ChangeMsgIdFn(msgID MsgIdFunction) {
```

Comment:
let's call this `SetMsgIdFn`, as it is always called to set it.

Reply:
:+1: See 6bd07a7